File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/Entities/CourseLesson.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.Entities
{
    public class CourseLesson : Entity
    {
        public Guid CourseLessonId { get; set; }
        public Guid CourseId { get; set; }
        public Guid LessonId { get; set; }
        public int Offset { get; set; }
        public string LessonTitle { get; set; } = string.Empty;
        protected override CourseLesson Init()
        {
            return new CourseLesson();
        }
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/entities/course-lesson.js

import Entity from './entity';
class CourseLesson extends Entity{
    constructor() {
        super();
        this.CourseLessonId = null;
        this.CourseId = null;
        this.LessonId = null;
        this.Offset = null;
        this.LessonTitle = null;
    }

    init() {
        return new CourseLesson();
    }
}

export default CourseLesson;